### PR TITLE
Added StepLR scheduler

### PR DESCRIPTION
**Pull request by miknyko**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

https://github.com/tracel-ai/burn/issues/1198

### Changes

Hi, I added the step lr according to the pytorch implementation.  I will add more LR schdulers later this week .
:D

### Testing

Testing unit added

### CODE REVIEW

**Review by nathanielsimard — APPROVED** [review]

File: burn-core/src/lr_scheduler/step.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    gamma: f64,
+}
+
+/// Decays the learning rate of each parameter group by gamma every step_size epochs.
+#[derive(Clone, Debug)]
+pub struct StepLrScheduler {
+    init_lr: LearningRate,
+    step_size: i64,
+    gamma: f64,
+    step: f64,
```
Comment (laggui) [inline burn-core/src/lr_scheduler/step.rs:27]: Any reason why the step is `f64` instead of unsigned int like `usize`?

**Review by laggui — CHANGES_REQUESTED** [review]
Overall the implementation looks good to me! Good job 👍 

Just have a few minor comments below.

File: burn-core/src/lr_scheduler/step.rs (hunk 1 above)
Comment (nathanielsimard) [reply burn-core/src/lr_scheduler/step.rs:27]: I think in the state, it's just to avoid casting.

Comment (laggui) [reply burn-core/src/lr_scheduler/step.rs:27]: But in `to_record` the step is actually cast as `usize`, and then in `load_record` cast back as `f64`. Maybe I'm missing something, but it just seems like we could keep it `usize` all the way and avoid casting, no?

Comment (nathanielsimard) [reply burn-core/src/lr_scheduler/step.rs:27]: But it's not casted in the `step` function 😅

Comment (nathanielsimard) [reply burn-core/src/lr_scheduler/step.rs:27]: We could use `usize` but I would also update it in the Noam scheduler for consistency.

Comment (laggui) [reply burn-core/src/lr_scheduler/step.rs:27]: > But it's not casted in the `step` function 😅

Not when incremented, but when it is actually used it seems that it is also casted:

```rust
self.gamma.powf((self.step as i64 / self.step_size) as f64)
```

> We could use `usize` but I would also update it in the Noam scheduler for consistency.

Ahhhhh so that's where it comes from! I didn't realize that was the case.

**Comment by miknyko** [comment]
Sorry guys, quite busy past two weeks, I will continue working on the schedulers

**Comment by laggui** [comment]
> Sorry guys, quite busy past two weeks, I will continue working on the schedulers

No worries 🙂
